src/components/HdIcon.vue

```diff
@@ -53,6 +65,25 @@ export default {
 
       return svg;
     },
+    addAccessibilityTags(svg) {
+      const randId = Math.floor(Math.random() * 1000);
+      if (this.title) {
+        const titleId = `${this.id || randId}-title`;
+        const titleTag = document.createElementNS('http://www.w3.org/2000/svg', 'title');
+        titleTag.textContent = this.title;
+        titleTag.id = titleId;
+        svg.insertBefore(titleTag, svg.firstChild);
+        svg.setAttribute('aria-labelledby', titleId);
+      }
```

Comment:
I would suggest you to split `title` and `description` into separated functions just as a matter of structuring the code.

Reply:
Maybe you can use `if` like that, this practice may be more readable

```js
if (this.title) return false;
const randId = Math.floor(Math.random() * 1000);
const titleId = `${this.id || randId}-title`;
const titleTag = document.createElementNS('http://www.w3.org/2000/svg', 'title');
titleTag.textContent = this.title;
titleTag.id = titleId;
svg.insertBefore(titleTag, svg.firstChild);
svg.setAttribute('aria-labelledby', titleId);

```